### PR TITLE
Integrated iOS-only `accessibilityLanguage` prop

### PR DESCRIPTION
**Pull request by dgopsq**
## Summary

This PR fixes https://github.com/facebook/react-native/issues/30891

This PR is going to add an `accessibilityLanguage` prop to all the available components. This props is currently working only on iOS and should follow the [guidelines of the relative configuration](https://developer.apple.com/documentation/objectivec/nsobject/1615192-accessibilitylanguage).

I'm in no way an expert on native programming (especially Objective-C) so I'm open to changes / improvements 🙂

## Changelog

[iOS] [Added] - Integrated the `accessibilityLanguage` prop to all the available components. The prop is available for any platform but it will work only on iOS.

## Test Plan

This has been tested using both the Simulator, checking for the `Language` attribute, and using a physical device with the Voice Over enabled.

Screenshot 2022-02-11 at 13 17 32

### CODE REVIEW

**Review by rh389 — COMMENTED** [review]
Hi @dgopsq and thanks for the PR - this is a great first contribution!

I think we should add a platform hint to the comments to indicate that it's only implemented in iOS, and a [docs PR](https://github.com/facebook/react-native-website/blob/main/docs/view.md) would also be great, but this looks on the right track to me.

File: Libraries/Components/View/ViewPropTypes.js [hunk 1]
```diff
@@ -415,6 +415,13 @@ export type ViewProps = $ReadOnly<{|
    */
   accessibilityHint?: ?Stringish,
 
+  /**
+   * Indicates to the accessibility services that the UI component is in
+   * a specific language. The provided string should be formatted following
+   * the BCP 47 specification (https://www.rfc-editor.org/info/bcp47).
+   */
```
Comment (rh389) [inline Libraries/Components/View/ViewPropTypes.js:422]:
```suggestion
   *
   * @platform ios
   */
```

**Comment by dgopsq** [comment]
> Hi @dgopsq and thanks for the PR - this is a great first contribution!
> 
> I think we should add a platform hint to the comments to indicate that it's only implemented in iOS, and a [docs PR](https://github.com/facebook/react-native-website/blob/main/docs/view.md) would also be great, but this looks on the right track to me.

Thank you @rh389! A PR for the docs will follow soon 🙂

**Review by PrayaAmadigaPitasa — COMMENTED** [review]
need documentation for new props

**Comment by dgopsq** [comment]
@PrayaAmadigaPitasa  I've been a bit busy, in the next days I'll open the PR :)

By the way, I just noticed that the latest tests didn't pass on the CI but I cannot reproduce anything locally, and I don't think I'm see useful informations about the error, do you know what's the matter?

**Comment by dgopsq** [comment]
I opened https://github.com/facebook/react-native-website/pull/2970 to address the documentation!

**Comment by dgopsq** [comment]
@rh389 @PrayaAmadigaPitasa Hello, sorry in advance for the notification, I just wanted to know if something else needs to be done on my part, if so let me know 🙂

**Comment by rh389** [comment]
Thanks @dgopsq, sorry for the delay. This looks good to me - imported for final review

**Comment by dvijeniii05** [comment]
Hi everyone! 
I ahve a strange issue, where none of default React Native elements, such as View, Text, Touchable etc, they don't list accessibilityLanguage as a prop. None of them. Every other accessibility prop is avaialble axcept Language. 

Currently tried on 3 different "machines", 3 different projects, even a new one out of the box, and nothing.
Is there an issue with it? has the prop been removed?

**Comment by gaurav5430** [comment]
> Hi everyone! I have a strange issue, where none of default React Native elements, such as View, Text, Touchable etc, they don't list accessibilityLanguage as a prop. None of them. Every other accessibility prop is available except Language.
> 
> Currently tried on 3 different "machines", 3 different projects, even a new one out of the box, and nothing. Is there an issue with it? has the prop been removed?

I have the same problem, is it something to do with react native version?

**Comment by gaurav5430** [comment]
@dgopsq how do you get the Advanced panel in accessibility inspector ?

**Comment by gaurav5430** [comment]
> > Hi everyone! I have a strange issue, where none of default React Native elements, such as View, Text, Touchable etc, they don't list accessibilityLanguage as a prop. None of them. Every other accessibility prop is available except Language.
> > Currently tried on 3 different "machines", 3 different projects, even a new one out of the box, and nothing. Is there an issue with it? has the prop been removed?
> 
> I have the same problem, is it something to do with react native version?

As i can see, this is available from version 69 onwards. I am still on 68

**Comment by dgopsq** [comment]
> @dgopsq how do you get the Advanced panel in accessibility inspector ?

@gaurav5430 I didn't do anything special, I just had it from the start. As far as I can see it's not even possible to hide it so you should have it too! 🤔

> Hi everyone! I have a strange issue, where none of default React Native elements, such as View, Text, Touchable etc, they don't list accessibilityLanguage as a prop. None of them. Every other accessibility prop is available except Language.
> 
> Currently tried on 3 different "machines", 3 different projects, even a new one out of the box, and nothing. Is there an issue with it? has the prop been removed?

@dvijeniii05 You're using TypeScript right? I just noticed that `@types/react-native` doesn't have the `accessibilityLanguage` prop. I'll try to make a PR as soon as possible 🙂

**Pull request closed without merging**